tests/test_upload.py

```diff
@@ -777,6 +816,75 @@ def test_update_new_balance_with_no_previous_balance(self, mock_get_connection):
         })
 
 
+class SettlementDateParsingTestCase(TestCase):
+    def test_parsable_settlement_2_digit_dates(self):
+        values = [
+            ('31', date(2021, 1, 31), date(2021, 1, 31)),
```

Comment:
Is this what you'd expect? I would have thought that `31` in this context (where relative date is 31st January 2021) would mean 31th December 2020 instead.

I guess it's related to my comment about the `<=` in these 2 utility function.

Reply:
i'm just leaving the logic that already existed: allowing for the settlement date to be the same as the date of the transaction/"today". this is indeed related to the `<=` vs `<` in parse_2_digit_date and parse_4_digit_date. i don't mind making it strictly less, but i thought it'd be safer to keep it as it was

Reply:
That's perfectly fine - You're probably right, maintain the existing behaviour is probably the safe/right thing to do so don't worry - and we can always change in the future if the need arises ☺️